### PR TITLE
k8s storage is opinionated model config

### PR DESCRIPTION
**Pull request by wallyworld**
## Description of change

Juju now uses opinionated storage defaults when provisioning storage on k8s. The main changes:
- add-k8s queries the cluster and prompts for storage if no suitable storage class can be found
- the k8s provider gains config support for storing the workload and operator storage classes as model default attributes
- a bunch of code used to search for k8s storage classes is deleted

## QA steps

Use add-k8s to import microk8s
check that model-defaults shows hostpath storage for operator and workload
add a model
check that model config shows hostpath storage for operator and workload
deploy a charm with storage
create a storage pool and deploy a charm with pool storage

### CODE REVIEW

File: caas/kubernetes/provider/init.go [hunk 1]
```diff
@@ -40,6 +41,12 @@ func init() {
 			Provisioner: "kubernetes.io/aws-ebs",
 		},
 	}
+
+	// clusterPreferredOperatorStorage defines the opinionated storage
+	// that Juju requires to be available on supported clusters to
+	// provision storage for operators.
+	// TODO - support regional storage for GCE etc
+	clusterPreferredOperatorStorage = clusterPreferredWorkloadStorage
```
Comment (ycliuhw) [inline caas/kubernetes/provider/init.go]: we probably should rename `clusterPreferredOperatorStorage` to `jujuPreferredStorage` for both app `operator` and `controller` pod later.

Comment (wallyworld) [reply caas/kubernetes/provider/init.go]: yeah, ok, will do that before landing

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
[extend cloud model to allow default storage class to be specified](https://trello.com/c/03FGMyWJ/645-extend-cloud-model-to-allow-default-storage-class-to-be-specified)

**Pull request merged**